python/tank/util/shotgun.py

```diff
@@ -236,12 +240,23 @@ def download_url(sg, url, location):
     
     # download the given url
     try:
+        request = urllib2.Request(url)
+        if sg.config.server in url:
+            request.add_header("user-agent", "; ".join(sg._user_agents))
+
         if timeout and sys.version_info >= (2,6):
             # timeout parameter only available in python 2.6+
-            response = urllib2.urlopen(url, timeout=timeout)
+            response = urllib2.urlopen(request, timeout=timeout)
         else:
             # use system default
-            response = urllib2.urlopen(url)
+            response = urllib2.urlopen(request)
+
+        if use_url_extension:
+            # Make sure the disk location has the same extension as the url path.
+            loc_base, loc_ext = os.path.splitext(location)
+            url_ext = os.path.splitext(urlparse.urlparse(response.geturl()).path)[-1]
+            if loc_ext != url_ext:
+                location = "%s%s" % (loc_base, url_ext)
```

Comment:
what happens if the url target doesn't have an extension in this case? if the url is `https://hostname/some_long_url`? Can we tweak the logic so that we only append an extension if there is one available?